src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,195 @@
+
+                    // TO-DO: If the change involves modifying any document besides the document where the code action
+                    // was invoked, temporarily execute via command instead of WorkspaceEdit until LSP bug is fixed:
+                    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After bug is fixed, remove the below if-statement and the existing code should work.
+                    if (changedDocuments.Any(d => d != document.Id) ||
+                        changedAnalyzerConfigDocuments.Any() ||
+                        changedAdditionalDocuments.Any())
+                    {
+                        runAsCommand = true;
+                        break;
+                    }
+
+                    // Changed documents
+                    await AddTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetDocument, solution.GetDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed analyzer config documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await AddTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedAnalyzerConfigDocuments,
+                        applyChangesOperation.ChangedSolution.GetAnalyzerConfigDocument, solution.GetAnalyzerConfigDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed additional documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await AddTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedAdditionalDocuments,
+                        applyChangesOperation.ChangedSolution.GetAdditionalDocument, solution.GetAdditionalDocument,
+                        cancellationToken).ConfigureAwait(false);
+                }
+
+                codeAction.Edit = new LSP.WorkspaceEdit { DocumentChanges = textDocumentEdits.ToArray() };
+            }
+
+            // Set up to run as command on the server instead of using WorkspaceEdits.
+            var commandOperations = operations.All(operation => !(operation is ApplyChangesOperation));
+            if (commandOperations || runAsCommand)
+            {
+                codeAction.Command = new LSP.Command
+                {
+                    CommandIdentifier = CodeActionsHandler.RunCodeActionCommandName,
+                    Title = codeAction.Title,
+                    Arguments = new object[] { data }
+                };
+            }
+
+            return codeAction;
```

Comment:
the code flow here is extremely weird.  let's chat tomorrow, i think this can be definitely simplified.